graphql_compiler/schema_transformation/utils.py

```diff
@@ -46,16 +46,62 @@ class SchemaStructureError(SchemaTransformError):
 
 
 class InvalidTypeNameError(SchemaTransformError):
-    """Raised if a type/field name is not valid.
+    """Raised if a type name is not valid.
 
-    This may be raised if the input schema contains invalid names, or if the user attempts to
-    rename a type/field to an invalid name. A name is considered valid if it consists of
-    alphanumeric characters and underscores and doesn't start with a numeric character (as
-    required by GraphQL), and doesn't start with double underscores as such type names are
+    This may be raised if the input schema contains invalid names. A name is considered valid if it
+    consists of alphanumeric characters and underscores and doesn't start with a numeric character
+    (as required by GraphQL), and doesn't start with double underscores as such type names are
     reserved for GraphQL internal use.
+
+    Not to be confused with SchemaRenameInvalidNameError, which is used for schemas being renamed
+    and includes more information.
     """
 
 
+class SchemaRenameInvalidNameError(SchemaTransformError):
+    """Raised if a type name is not valid during renaming.
+
+    A name is considered valid if it consists of alphanumeric characters and underscores and doesn't
+    start with a numeric character (as required by GraphQL), and doesn't start with double
+    underscores as such type names are reserved for GraphQL internal use.
+
+    Not to be confused with InvalidTypeNameError, which is used for schemas that aren't being
+    renamed because this type of error includes information from the renaming process that
+    InvalidTypeNameError doesn't.
+    """
+
+    # TODO(Leon): Would like feedback on this-- is there a better way to do this than have two types
+    # of invalid name errors? See docstring for why I split them up.
```

Comment:
I actually didn't get what the docstring was saying. Can you maybe give an example to clarify the distinction? "includes information that the other error doesn't" is not the most clear statement.

Reply:
`InvalidTypeNameError` used to have double duty: one when checking that a given schema has no invalid names (which is something that happens in code unrelated to schema renaming), and one to check that schema renaming doesn't produce a schema with invalid names.

My goal here was to make a clear error if schema renaming involves invalid names, which means tracking the problem renamings (perhaps a dict mapping valid names in the original schema to the invalid names in the renamed schema). The problem is, this sort of bookkeeping doesn't make sense when the code raises `InvalidTypeNameError` when just checking a given schema unrelated to schema renaming specifically.

This extra information means that we have two errors that do different things despite having similar names.

Reply:
I don't think we should add the `SchemaRenameInvalidNameError` error class, specially as part of this PR, for the following reasons:
1. This PR renames `renamings` to `type_renamings` so that we can also add a `field_renamings` 
     field or something akin to that in the future. The addition of this class seems completely unrelated to that goal 
     and PRs should be aiming to do one thing and one thing only.  
2. The addition of this error class is only arguably an improvement to the code and I think we have more 
     important stuff to work on. I understand the goal you are trying to achieve here, but there is a downside to 
     having an excessive amount of error classes, the `InvalidTypeNameError` seemed perfectly fine to me and 
     the distinction between `SchemaRenameInvalidNameError` and `InvalidTypeNameError` is confusing. So 
     even though you might see this as an improvement, it is really not obvious to me that it is.